podaac/subsetter/subset.py

```diff
@@ -700,6 +698,10 @@ def build_cond(str_timestamp, compare):
                 epoch_datetime = dataset[epoch_time_var_name].values[0]
                 timestamp = np.datetime64(timestamp) - epoch_datetime
 
+        if np.issubdtype(dataset[time_var_name].dtype, np.dtype(float)):
+            start_date = np.datetime64(dataset[time_var_name].attrs['Units'][-10:])
+            timestamp = (np.datetime64(timestamp) - start_date).astype('timedelta64[s]').astype('float')
```

Comment:
No idea since OMI is the first one to have type float. Xarray isn't decoding the time, and the only piece of info to reference tai93 time is in the attributes. An option could be to creating separate logic to catch OMI and if xarray can't handle he5 time decoding the logic to the others it should be very similar to OMI - I put type float here because I'm assuming those future products will behave very similarly